ITRStartTask/AppLibrary/ApiService/Operations/FeedItemDownloadAsyncOperation.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  FeedItemDownloadOperation2.swift
+//  ITRStartTask
+//
+//  Created by Homyakov, Ilya2 on 23/08/2018.
+//  Copyright © 2018 Homyakov, Ilya2. All rights reserved.
+//
+
+import Foundation
+
+class FeedItemDownloadAsyncOperation<T: Codable>: AsyncOperation {
+    enum FeedItemDownloadAsyncOperationError: Int {
+        case not200
+        case incorrectData
+        case operationCanceled
+
+        func errorCode() -> Int {
+            return self.rawValue
+        }
+
+        func error() -> Error {
+            switch self {
+            case .not200:
+                let errorString = NSLocalizedString("MessageFailToConnect", comment: "")
+                return NSError.init(domain: "ServiceClientError",
+                                    code: FeedItemDownloadAsyncOperationError.not200.errorCode(),
+                                    userInfo: ["localizedDescription": errorString])
+            case .incorrectData:
+                let localizedDescription = NSLocalizedString("IncorrectServerData", comment: "")
+                return NSError.init(domain: "FeedItemDownloadAsyncOperationError",
+                                    code: FeedItemDownloadAsyncOperationError.incorrectData.errorCode(),
+                                    userInfo: ["localizedDescription": localizedDescription])
+            case .operationCanceled:
+                let localizedDescription = NSLocalizedString("OperationCanceled", comment: "")
+                return NSError.init(domain: "FeedItemDownloadAsyncOperationError",
+                                    code: FeedItemDownloadAsyncOperationError.operationCanceled.errorCode(),
+                                    userInfo: ["localizedDescription": localizedDescription])
+            }
+        }
+    }
+
+    private let deserializer: FeedItemDeserializerProtocol = FeedItemDeserializer ()
+
+    var url: URL
+    var storyItem: T?
+    var error: Error?
+
+    init (url: URL) {
+        self.url = url
+        super.init()
+    }
+
+    override func main() {
+        if isCancelled {
+            return
+        }
+
+        getFeedItem(url: self.url, onSuccess: { [weak self] (feedItem: T) in
+            self?.storyItem = feedItem
+            self?.state = .finished
+        }, onFailure: { [weak self] (error) in
+            self?.error = error
+            self?.state = .finished
+        })
+    }
+
+    private func getFeedItem<T: Codable>(url: URL,
+                                         onSuccess: @escaping (T) -> Void,
+                                         onFailure: @escaping (Error) -> Void) {
+        let task = URLSession.shared.dataTask(with: url) {(data, response, error) in
+            if let error = error {
+                onFailure(error)
+
```

Comment:
Empty line should not be used for this kind of short blocks of code

Reply:
fixed